apps/cli/src/code-generator.ts

```diff
@@ -0,0 +1,216 @@
+import { Tree, formatFiles, generateFiles } from '@nx/devkit'
+import { printChanges } from 'nx/src/command-line/generate/generate'
+import { FsTree, flushChanges } from 'nx/src/generators/tree'
```

Comment:
I don't think the CLI should rely on NX. 

The CLI we should be able to install and use without any dependency on NX. Then we can have a NX plugin which calls the CLI, or imports it as a library and uses it.

Reply:
Can I have a chat about this one I'm not really happy with how I've implemented this in general.

Reply:
Updated as per chat